gateway/src/apicast/proxy.lua

```diff
@@ -313,65 +309,63 @@ local function response_codes_data()
   return params
 end
 
-local function post_action(_, self, cached_key, backend, ...)
+local function handle_timer_post_action(_, self, timers, cached_key, backend, ...)
   local res = util.timer('backend post_action', backend.authrep, backend, ...)
 
-  if not using_post_action then
+  if not using_sync_post_action then
     timers:post(1)
   end
 
   self:handle_backend_response(cached_key, res)
 end
 
-local function capture_post_action(self, cached_key, service)
+local function sync_post_action(self, cached_key, service, credentials, formatted_usage)
   local backend = assert(backend_client:new(service, http_ng_ngx), 'missing backend')
-  local formatted_usage = format_usage(self.usage)
   local res = backend:authrep(
     formatted_usage,
-    self.credentials,
+    credentials,
     response_codes_data(),
     self.extra_params_backend_authrep
   )
 
   self:handle_backend_response(cached_key, res)
 end
 
-local function timer_post_action(self, cached_key, service)
+local function async_post_action(self, cached_key, service, credentials, formatted_usage)
   local backend = assert(backend_client:new(service), 'missing backend')
 
   local ok, err = timers:wait(10)
 
   if ok then
     -- TODO: try to do this in different phase and use semaphore to limit number of background threads
     -- TODO: Also it is possible to use sets in shared memory to enqueue work
-    local formatted_usage = format_usage(self.usage)
-    ngx.timer.at(0, post_action, self, cached_key, backend, formatted_usage, self.credentials, response_codes_data())
+
+    ngx.timer.at(0, handle_timer_post_action, self, timers, cached_key, backend, formatted_usage, credentials, response_codes_data())
```

Comment:
@davidor looks like this is missing `self.extra_params_backend_authrep`

Reply:
Good catch :+1: 

This means that the referrer filter policy does not work when using the experimental async reporting. I'll add a test and fix the issue.